### ISSUE TITLE
Invalid handler specified when loading integration

### ISSUE THREAD

**Issue by karlinurswick** [asantaga/wiserHomeAssistantPlatform#537]
I have had the wiser integration working fine in HomeAssistant working within a VirtualBox machine on my normal use laptop for some time and have now moved to a dedicated machine for home assistant. the new machine is a similar setup, runs windows 10 with HomeAssistant operating in VirtualBox Linux machines as before.  I have installed HACS and then subsequently downloaded The Wiser package.  When trying t add this via the "Add Integration" Button I get an error - Config flow could not be loaded: {"Invalid handler specified"} and the integration will not load.

When checking the logs, I see this;
Logger: homeassistant.config_entries
Source: config_entries.py:3249
First occurred: 7:19:29 AM (3 occurrences)
Last logged: 7:33:21 AM
Error occurred loading flow for integration wiser: No module named 'aioWiserHeatAPI'

I have searched via google for a couple of evenings and cant find any way to resolve this, can anybody offer any advice?  Thanks.

**Comment by msp1974** [comment]
What version of HA are you running and by what method? HAOS, docker etc?

**Comment by karlinurswick** [comment]
I am running version 13.2, core 2024.11.3, supervisor 2024.11.4 and believe this is a supervised installation in a Linux virtual (virtualbox) machine.  Followed the instructions from here - https://www.home-assistant.io/installation/windows/

**Comment by karlinurswick** [comment]
Not sure how much of this is needed so have added all the log from where the wiser issue starts, thanks for taking the time to look at this.
2024-12-03 17:04:05.492 ERROR (MainThread) [homeassistant.config_entries] Error occurred loading flow for integration wiser: No module named 'aioWiserHeatAPI'


2024-12-03 17:04:05.493 ERROR (MainThread) [homeassistant] Error doing job: Task exception was never retrieved (None)

Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 3247, in _load_integration

await integration.async_get_platform("config_flow")
  File "/usr/src/homeassistant/homeassistant/loader.py", line 1104, in async_get_platform
    
platforms = await self.async_get_platforms((platform_name,))
                
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/loader.py", line 1181, in async_get_platforms
    
import_future.result()
  File "/usr/src/homeassistant/homeassistant/loader.py", line 1151, in async_get_platforms
    
await self.hass.async_add_import_executor_job(
  File "/usr/local/lib/python3.12/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
             
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/loader.py", line 1094, in _load_platforms
    platform_name: self._load_platform(platform_name)
                   
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/loader.py", line 1268, in _load_platform
    cache[full_name] = self._import_platform(platform_name)
                       
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/loader.py", line 1300, in _import_platform
    return importlib.import_module(f"{self.pkg_path}.{platform_name}")
           
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/util/loop.py", line 200, in protected_loop_func
    return func(*args, **kwargs)
           
^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/importlib/__init__.py", line 90, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen importlib._bootstrap>", line 1387, in _gcd_import
  File "<frozen importlib._bootstrap>", 
line 1360, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1310, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 488, 
in _call_with_frames_removed
  File "<frozen importlib._bootstrap>", line 1387, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1360, in _find_and_load
  File "<frozen importlib._bootstrap>", 
line 1331, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 935, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 995, 
in exec_module
  File "<frozen importlib._bootstrap>", line 488, in _call_with_frames_removed
  File "/config/custom_components/wiser/__init__.py", line 23, in <module>
    
from .coordinator import WiserUpdateCoordinator
  File "/config/custom_components/wiser/coordinator.py", line 8, in <module>
    from aioWiserHeatAPI.wiserhub import 
(
ModuleNotFoundError: No module named 'aioWiserHeatAPI'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  
File "/usr/src/homeassistant/homeassistant/helpers/discovery_flow.py", line 137, in _async_start
    await gather_with_limited_concurrency(FLOW_INIT_LIMIT, *init_coros)
  
File "/usr/src/homeassistant/homeassistant/util/async_.py", line 114, in gather_with_limited_concurrency
    return await gather(
           
^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/util/async_.py", line 112, in sem_task
    return await task
           
^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 1301, in async_init
    flow, result = await self._async_init(flow_id, handler, context, data)
                   
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 1326, in _async_init
    
flow = await self.async_create_flow(handler, context=context, data=data)
           
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 1518, in async_create_flow
    
handler = await _async_get_flow_handler(
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/config_entries.py", 
line 3268, in _async_get_flow_handler
    await _load_integration(hass, domain, hass_config)
  
File "/usr/src/homeassistant/homeassistant/config_entries.py", line 3254, in _load_integration
    raise data_entry_flow.UnknownHandler from err
homeassistant.data_entry_flow.UnknownHandler

2024-12-03 17:04:50.603 ERROR (MainThread) [homeassistant.config_entries] Error occurred loading flow for integration wiser: No module named 'aioWiserHeatAPI'

2024-12-03 17:11:58.071 ERROR (MainThread) [homeassistant.config_entries] Error occurred loading flow for integration wiser: No module named 'aioWiserHeatAPI'

**Comment by msp1974** [comment]
OK, if there is nothing there then probably best approach is to access the shell in the HA docker and try and install it manually and see if that gives an error.

Get into shell by doing below from Advanced SSH & Web Terminal
```
docker exec -it homeassistant bash
```

now install with
```
pip install aioWiserHeatAPI
```

see what it does - either it will install and then stop HA complaining or it will error and we will know the issue.

**Comment by msp1974** [comment]
Wierdly, I just got a notification about exactly this issue but across many HACs installs for newly installed HA environments (new from 2024.10 onwards).  I think this is your issue.

https://github.com/home-assistant/core/issues/127966

**Comment by karlinurswick** [comment]
Thank you for looking into this, I have got it working now.  I deleted the virtual machine and started a new one with a fresh download of homeassistant, once started I turned on advanced mode, installed the add-on "Terminal and SSH" then used this to download HACS.  once i had the HACS integration running, I then downloaded and installed the wiser integration which has worked straight away.  In the previous attempt, I had downloaded HACS using the installer link on the HACS website, not sure if this has mad any difference or whether it is due to having downloaded a fresh version of HA this morning.  Thanks again.
